In [1]:
import pandas as pd
import numpy as np


In [2]:
mileage_0501=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/05mileageReport_Part_1.csv')
mileage_0502=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/05mileageReport_Part_2.csv')
mileage_0503=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/05mileageReport_Part_3.csv')
mileage_0504=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/05mileageReport_Part_4.csv')
mileage_05=pd.concat([mileage_0501,mileage_0502,mileage_0503,mileage_0504])

In [3]:
mileage_05.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1651862 entries, 0 to 151861
Data columns (total 7 columns):
 #   Column               Non-Null Count    Dtype  
---  ------               --------------    -----  
 0   Unit                 1651862 non-null  object 
 1   Time/Date (COT)      1651862 non-null  object 
 2   Address              1651862 non-null  object 
 3   Distance (km)        1651862 non-null  float64
 4   Total Distance (km)  1651862 non-null  float64
 5   Odometer             1651862 non-null  float64
 6   Driver               0 non-null        float64
dtypes: float64(4), object(3)
memory usage: 100.8+ MB


In [4]:
mileage_05['Distance (km)'].value_counts(ascending=False)

Distance (km)
0.10     610703
0.20     323185
0.30     193620
0.40     125743
0.50      82449
          ...  
11.94         1
15.37         1
56.78         1
41.47         1
8.14          1
Name: count, Length: 2616, dtype: int64

In [5]:
mileage_05['Distance (km)'].sort_values(ascending=False)

263879    163905.58
13132     148180.38
210544    144435.53
263650     81702.09
212678     74385.86
            ...    
217557         0.00
174978         0.00
193713         0.00
219082         0.00
20394          0.00
Name: Distance (km), Length: 1651862, dtype: float64

In [6]:
for i in range(0,1651862):
    if mileage_05.iloc[i,3]>=1530:
        print(mileage_05.iloc[i,0],mileage_05.iloc[i,1],mileage_05.iloc[i,3],mileage_05.iloc[i,4])

EQY796 06/05/2025 9:10 AM 59516.31 60004.94
NNZ794 15/05/2025 1:11 AM 28881.77 0.0
EQZ347 05/05/2025 11:43 AM 11713.81 11726.47
EQZ347 06/05/2025 11:14 AM 11805.22 23570.12
EQZ347 06/05/2025 4:46 PM 11825.52 35415.34
EQZ347 10/05/2025 5:01 PM 12069.53 0.0
EQZ347 13/05/2025 8:46 AM 12169.93 0.0
EYX839 12/05/2025 7:37 PM 22846.3 0.0
NNZ787 14/05/2025 2:30 PM 30908.16 0.0
NNZ787 15/05/2025 10:20 AM 31062.46 0.0
TDS986 02/05/2025 10:41 AM 64691.53 64786.64
TDS986 05/05/2025 8:05 AM 65153.66 131673.38
TDS986 10/05/2025 10:29 AM 66097.8 0.0
TDS986 11/05/2025 8:50 AM 66415.51 0.0
TDS986 12/05/2025 12:44 PM 66629.32 0.0
TDS986 12/05/2025 6:01 PM 66782.63 0.0
TDS986 13/05/2025 5:06 PM 66925.34 0.0
TDS986 14/05/2025 11:06 AM 67058.44 0.0
EYX831 03/05/2025 8:54 AM 30648.45 30692.82
ERK706 01/05/2025 1:29 PM 16812.44 16818.55
VKQ018 03/05/2025 7:56 AM 25837.13 27371.06
VKQ018 05/05/2025 10:19 AM 26096.25 53623.0
VKQ018 05/05/2025 12:58 PM 52249.32 105885.75
VKQ018 05/05/2025 5:14 PM 26138.14 13203

In [7]:
wou053=mileage_05[mileage_05['Unit'].str.contains('WOU053')]
wou053['Distance (km)'].sort_values(ascending=False)

122763    61043.58
122415    60971.78
117214    59234.20
116824    59118.89
116511    58988.89
            ...   
119400        0.10
121803        0.10
116310        0.08
116360        0.02
116218        0.02
Name: Distance (km), Length: 8382, dtype: float64

In [8]:
mileage_0601=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/06mileageReport_Part_1.csv')
mileage_0602=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/06mileageReport_Part_2.csv')
mileage_0603=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/06mileageReport_Part_3.csv')
mileage_0604=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/06mileageReport_Part_4.csv')
mileage_06=pd.concat([mileage_0601,mileage_0602,mileage_0603,mileage_0604])

In [9]:
mileage_06.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1662036 entries, 0 to 162035
Data columns (total 7 columns):
 #   Column               Non-Null Count    Dtype  
---  ------               --------------    -----  
 0   Unit                 1662036 non-null  object 
 1   Time/Date (COT)      1662036 non-null  object 
 2   Address              1662036 non-null  object 
 3   Distance (km)        1662036 non-null  float64
 4   Total Distance (km)  1662036 non-null  float64
 5   Odometer             1662036 non-null  float64
 6   Driver               0 non-null        float64
dtypes: float64(4), object(3)
memory usage: 101.4+ MB


In [10]:
mileage_06['Distance (km)'].value_counts(ascending=False)

Distance (km)
0.10    615043
0.20    339292
0.30    203102
0.40    128125
0.50     83335
         ...  
7.11         1
2.47         1
1.47         1
6.89         1
3.01         1
Name: count, Length: 979, dtype: int64

In [11]:
mileage_06['Distance (km)'].sort_values(ascending=False)

108410    4113.99
149083    1531.56
474992     849.44
144357     791.73
144355     695.33
           ...   
332284       0.00
117782       0.00
376823       0.00
130125       0.00
140269       0.00
Name: Distance (km), Length: 1662036, dtype: float64

In [12]:
for i in range(0,1662036):
    if mileage_06.iloc[i,3]>=1530:
        print(mileage_06.iloc[i,0],mileage_06.iloc[i,1],mileage_06.iloc[i,3])

WOU053 01/06/2025 2:27 PM 4113.99
NNZ903 26/06/2025 12:42 PM 1531.56


In [13]:
wou053=mileage_06[mileage_06['Unit'].str.contains('WOU053')]
wou053

Unit     Time/Date (COT)                    Address  Distance (km)  \
108410  WOU053  01/06/2025 2:27 PM  (No street address given)        4113.99   
108411  WOU053  01/06/2025 2:27 PM  (No street address given)           0.60   
108412  WOU053  01/06/2025 2:28 PM  (No street address given)           0.60   
108413  WOU053  01/06/2025 2:28 PM  (No street address given)           1.10   
108414  WOU053  01/06/2025 2:29 PM  (No street address given)           0.10   
...        ...                 ...                        ...            ...   
127518  WOU053  29/06/2025 9:27 PM  (No street address given)           0.30   
127519  WOU053  29/06/2025 9:27 PM  (No street address given)           0.40   
127520  WOU053  29/06/2025 9:28 PM  (No street address given)           0.30   
127521  WOU053  29/06/2025 9:28 PM  (No street address given)           0.10   
127522  WOU053  29/06/2025 9:29 PM  (No street address given)           0.70   

        Total Distance (km)  Odometer  Driver  
108410              4113.99  65544.48     NaN  
108411              4114.59  65545.07     NaN  
108412              4115.19  65545.68     NaN  
108413              4116.29  65546.78     NaN  
108414              4116.39  65546.87     NaN  
...                     ...       ...     ...  
127518             10697.18  71931.56     NaN  
127519             10697.58  71931.96     NaN  
127520             10697.88  71932.25     NaN  
127521             10697.98  71932.35     NaN  
127522             10698.68  71933.06     NaN  

[19113 rows x 7 columns]

In [14]:
mileage_0701=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/07mileageReport_Part_1.csv')
mileage_0702=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/07mileageReport_Part_2.csv')
mileage_0703=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/07mileageReport_Part_3.csv')
mileage_07=pd.concat([mileage_0701,mileage_0702,mileage_0703])

In [14]:
mileage_07.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1007799 entries, 0 to 7798
Data columns (total 7 columns):
 #   Column               Non-Null Count    Dtype  
---  ------               --------------    -----  
 0   Unit                 1007799 non-null  object 
 1   Time/Date (COT)      1007799 non-null  object 
 2   Address              1007799 non-null  object 
 3   Distance (km)        1007799 non-null  float64
 4   Total Distance (km)  1007799 non-null  float64
 5   Odometer             1007799 non-null  float64
 6   Driver               0 non-null        float64
dtypes: float64(4), object(3)
memory usage: 61.5+ MB


In [15]:
mileage_07['Distance (km)'].value_counts(ascending=False)

Distance (km)
0.10     356110
0.20     201042
0.30     123196
0.40      78531
0.50      52530
          ...  
13.07         1
12.13         1
6.52          1
26.69         1
9.68          1
Name: count, Length: 1184, dtype: int64

In [16]:
mileage_07['Distance (km)'].sort_values(ascending=False)

6482      1114.14
188760     926.24
484171     915.12
484173     912.72
484177     912.72
           ...   
47744        0.00
29545        0.00
200664       0.00
498535       0.00
223278       0.00
Name: Distance (km), Length: 1007799, dtype: float64

In [17]:
for i in range(0,1007799):
    if mileage_07.iloc[i,3]>=900:
        print(mileage_07.iloc[i,0],mileage_07.iloc[i,1],mileage_07.iloc[i,3])

NNZ902 12/07/2025 12:18 PM 912.64
WNM385 12/07/2025 7:29 AM 926.24
NNZ903 06/07/2025 5:19 PM 909.37
NNZ903 06/07/2025 5:24 PM 915.12
NNZ903 06/07/2025 5:28 PM 912.72
NNZ903 06/07/2025 5:36 PM 912.72
PRZ058 12/07/2025 6:46 AM 1114.14


In [18]:
nnz903=mileage_07[mileage_07['Unit'].str.contains('NNZ903')]
nnz903_0607=nnz903[nnz903['Time/Date (COT)'].str.contains('06/07/2025 5:')]
nnz903_0607.head(30)
nnz903_0607.to_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/NNZ903_dates.csv')

In [19]:
wnm385=mileage_07[mileage_07['Unit'].str.contains('WNM385')]
wnm385_1207=wnm385[wnm385['Time/Date (COT)'].str.contains('12/07/2025 7:2')]
wnm385_1207

Unit     Time/Date (COT)                    Address  Distance (km)  \
188749  WNM385  12/07/2025 7:20 AM  (No street address given)           0.80   
188750  WNM385  12/07/2025 7:21 AM  (No street address given)           1.10   
188751  WNM385  12/07/2025 7:22 AM  (No street address given)           1.10   
188752  WNM385  12/07/2025 7:23 AM  (No street address given)           1.10   
188753  WNM385  12/07/2025 7:24 AM  (No street address given)           0.80   
188754  WNM385  12/07/2025 7:25 AM  (No street address given)           0.30   
188755  WNM385  12/07/2025 7:26 AM  (No street address given)           0.20   
188756  WNM385  12/07/2025 7:26 AM  (No street address given)           0.20   
188757  WNM385  12/07/2025 7:26 AM  (No street address given)           1.00   
188758  WNM385  12/07/2025 7:27 AM  (No street address given)           0.80   
188759  WNM385  12/07/2025 7:28 AM  (No street address given)           0.60   
188760  WNM385  12/07/2025 7:29 AM  (No street address given)         926.24   

        Total Distance (km)  Odometer  Driver  
188749              1660.47  30167.13     NaN  
188750              1661.57  30168.22     NaN  
188751              1662.67  30169.33     NaN  
188752              1663.77  30170.43     NaN  
188753              1664.57  30171.23     NaN  
188754              1664.87  30171.52     NaN  
188755              1665.07  30171.73     NaN  
188756              1665.27  30171.92     NaN  
188757              1666.27  30172.92     NaN  
188758              1667.07  30173.72     NaN  
188759              1667.67  30174.32     NaN  
188760              2593.91  31100.56     NaN

In [20]:
wnm385=mileage_07[mileage_07['Unit'].str.contains('WNM385')]
wnm385_1207=wnm385[wnm385['Time/Date (COT)'].str.contains('12/07/2025 7:3')]
wnm385_1207

Empty DataFrame
Columns: [Unit, Time/Date (COT), Address, Distance (km), Total Distance (km), Odometer, Driver]
Index: []

In [21]:
cu=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/Customer-Units.csv',sep=';',encoding='latin1')
cu.info()
cu.head(3)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 283 entries, 0 to 282
Data columns (total 47 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       283 non-null    int64  
 1   Owner                    283 non-null    object 
 2   Unit ID                  283 non-null    object 
 3   Unit                     283 non-null    object 
 4   Peripherals              0 non-null      float64
 5   Reporting Status         0 non-null      float64
 6   Show Until               0 non-null      float64
 7   Billing Group            4 non-null      object 
 8   Last Event Date          283 non-null    object 
 9   Monthly Fee              283 non-null    int64  
 10  Bill To                  20 non-null     object 
 11  Assigned To              283 non-null    object 
 12  Linked Mode              283 non-null    int64  
 13  Paired GPS               0 non-null      float64
 14  Paired Camera            0

ID    Owner Unit ID    Unit  Peripherals  Reporting Status  \
0  1105648  Renting    15JU  YAR184          NaN               NaN   
1  1105304  Renting    15AA  WOY701          NaN               NaN   
2  1106491   Client    16GV  WOY700          NaN               NaN   

   Show Until Billing Group Last Event Date  Monthly Fee  ...  \
0         NaN           NaN      21/07/2025        42800  ...   
1         NaN           NaN      18/07/2025        42800  ...   
2         NaN    INTERNAL U      21/07/2025       298600  ...   

     Assigned to Company IFTA  History Months  Comments  \
0  10/19/2024 3:28:30 PM    N               3       NaN   
1  10/26/2024 1:41:55 PM    N               3       NaN   
2  12/20/2024 9:33:11 AM    N               3       NaN   

        Warranty Expires Script JBOX Profile Contract Type Subscription  \
0   9/4/2025 12:00:00 AM    NaN          NaN           NaN          NaN   
1   9/4/2025 12:00:00 AM    NaN          NaN           NaN          NaN   
2  9/17/2025 12:00:00 AM    NaN          NaN           NaN          NaN   

  Firmware  
0      NaN  
1      NaN  
2      NaN  

[3 rows x 47 columns]

In [22]:
trouble_units_data=[['NNZ903'],['WNM385'],['PRZ058']]
trouble_units_df=pd.DataFrame(trouble_units_data,columns=['Shortname'])
trouble_units_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 1 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Shortname  3 non-null      object
dtypes: object(1)
memory usage: 156.0+ bytes


In [23]:
df_book_table01=trouble_units_df
df_book_table02=cu

In [26]:
num_filas_table01=int(df_book_table01.shape[0])
num_filas_table02=int(df_book_table02.shape[0])       
cont=0
df_aux=pd.DataFrame()        
list_field01=[]
list_field02=[]
list_field03=[]
list_field04=[]       
#print("df_book_table01[8,0]: ",df_book_table01.iloc[8,0]) #UnitsByDealer.csv_r.csv
#print("df_book_table02[8,5]: ",df_book_table02.iloc[8,5]) #6MB.        
print("\nInformación de las tablas: ")
print("\nTabla 1:")
print(df_book_table01.info())
print("\nTabla 2:")
print(df_book_table02.info())

attribute_name_table01='Shortname'#Digitar el índice de la tabla 1.input("\nDigita el atributo a comparar de la tabla 1: ")
attribute_name_table02='Unit'#Digitar el índice de la tabla 2.input("Digita el atribito a comparar de la tabla 2: ")

position_table01=int(df_book_table01.columns.get_loc(attribute_name_table01))
position_table02=int(df_book_table02.columns.get_loc(attribute_name_table02))
print("\nAtributo de la tabla1: ",position_table01)
print("Atributo de la tabla2: ",position_table02,"\n")
print("Trabajando...")
                          
for i in range(0,num_filas_table01):
    for j in range(0,num_filas_table02):
        #if str(df_book_table01.iloc[i,5])==str(df_book_table02.iloc[j,3]): 
        if str(df_book_table01.iloc[i,position_table01])==str(df_book_table02.iloc[j,position_table02]):                                                                             
            list_field01.append(df_book_table02.iloc[j,3])
            list_field02.append(df_book_table02.iloc[j,17])
            #list_field03.append(df_book_table02.iloc[j,4])
            #list_field04.append(df_book_table02.iloc[j,1])
            print("df_book_table01[i,position_table01]: ",df_book_table01.iloc[i,position_table01]," = ",end="")
            print("df_book_table02[j,position_table02]: ",df_book_table02.iloc[j,position_table02])               
            cont=cont+1
df_aux['Short name']=list_field01
df_aux['Unit type']=list_field02
#df_aux['FIELD03']=list_field03
#df_aux['FIELD04']=list_field04
#print("Cont= ",cont)
#print(df_aux.info())
print(df_aux)
option_exp=input("\n¿Desea generar un reporte de coincidencias en formato csv S/N?\n")
if option_exp=="S":
    df_aux.to_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/report01.csv')
cont=0
option_exp=input("¿Desea generar un reporte de diferencias en formato csv S/N?\n")        
df_book_table02['ACTION']="Keep"
if option_exp=="S":
    print("Trabajando...")
    for i in range(0,num_filas_table01):
        for j in range(0,num_filas_table02):
            if str(df_book_table01.iloc[i,position_table01])==str(df_book_table02.iloc[j,position_table02]):
                df_book_table02.iloc[j,5]=np.nan
                print("df_book_table01[i,position_table01]: ",df_book_table01.iloc[i,position_table01]," = ",end="")
                print("df_book_table02[j,position_table02]: ",df_book_table02.iloc[j,position_table02])               
    df_book_table02.dropna(axis=0,inplace=True)
    #print(df_book_table02)
    df_book_table02.to_csv("/home/jlhb1984/py-projects/data/chilcoTS000153784/report02.csv")

#lista_combinada=[lista_simcard,lista_linea]


Información de las tablas: 

Tabla 1:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 1 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Shortname  3 non-null      object
dtypes: object(1)
memory usage: 156.0+ bytes
None

Tabla 2:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 283 entries, 0 to 282
Data columns (total 47 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       283 non-null    int64  
 1   Owner                    283 non-null    object 
 2   Unit ID                  283 non-null    object 
 3   Unit                     283 non-null    object 
 4   Peripherals              0 non-null      float64
 5   Reporting Status         0 non-null      float64
 6   Show Until               0 non-null      float64
 7   Billing Group            4 non-null      object 
 8   Last Event Date          283 non-n

Nuevo análisis del Mileage Report 10-09-2025.

In [41]:
mr1=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/mileageReport_Part_1_2025-09-10_1757513035919_ic1lz280i.csv')
mr2=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/mileageReport_Part_2_2025-09-10_1757513059576_9fthfejt6.csv')
mr1.info()
mr2.info()
mr12=pd.concat([mr1,mr2])
mr12.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500000 entries, 0 to 499999
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   Unit                 500000 non-null  object 
 1   Time/Date (COT)      500000 non-null  object 
 2   Address              500000 non-null  object 
 3   Distance (km)        500000 non-null  float64
 4   Total Distance (km)  500000 non-null  float64
 5   Odometer             500000 non-null  float64
 6   Driver               0 non-null       float64
dtypes: float64(4), object(3)
memory usage: 26.7+ MB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23830 entries, 0 to 23829
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Unit                 23830 non-null  object 
 1   Time/Date (COT)      23830 non-null  object 
 2   Address              23830 non-null  object 
 3   Distance (km)  

In [39]:
mr12['Distance (km)'].sort_values(ascending=False)

431791    1165.43
431844     591.99
82708      539.92
261228     533.00
431847     498.48
           ...   
261802       0.00
262438       0.00
262396       0.00
262408       0.00
402775       0.00
Name: Distance (km), Length: 523830, dtype: float64

In [59]:
for i in range(0,523830):
    if mr12.iloc[i,3]>=200:
        print(mr12.iloc[i,0],mr12.iloc[i,1],mr12.iloc[i,3],mr12.iloc[i,5])

EYX834 09/03/2025 4:30 PM 248.11 30607.64
EYX834 09/03/2025 5:47 PM 237.01 30607.54
EYX840 09/02/2025 11:25 AM 261.91 66762.83
EYX840 09/02/2025 11:25 AM 539.92 67040.84
PRZ607 09/08/2025 12:33 PM 533.0 2933.83
UPO081 09/09/2025 11:09 AM 1165.43 36362.01
UPO081 09/09/2025 1:40 PM 591.99 37146.74
UPO081 09/09/2025 1:44 PM 486.04 37703.33
UPO081 09/09/2025 1:46 PM 498.48 38201.82
VKQ018 09/05/2025 5:48 PM 295.21 42820.27


In [44]:
mr12['Distance (km)'].max()

np.float64(1165.43)

In [45]:
vf=pd.read_csv('/home/jlhb1984/py-projects/data/chilcoTS000153784/rawdata_2bd4135b-58d1-4e3a-a48c-54cd5535ee14.csv')
vf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 503 entries, 0 to 502
Data columns (total 50 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Row Id                     503 non-null    int64  
 1   Sequence #                 0 non-null      float64
 2   Server Time                503 non-null    object 
 3   Unit Time                  503 non-null    object 
 4   Last Message Time          503 non-null    object 
 5   Latitude                   503 non-null    float64
 6   Longitude                  503 non-null    float64
 7   Driver Name                0 non-null      float64
 8   Event Name                 503 non-null    object 
 9   Raw Event                  503 non-null    object 
 10  Address                    503 non-null    object 
 11  GPS Speed                  393 non-null    float64
 12  ECU Speed                  0 non-null      float64
 13  Speed Limit                242 non-null    float64

In [46]:
vf['Distance'].max()

np.float64(54.64957945)

In [49]:
vf['Distance'].sort_values(ascending=False)

0      54.649579
1      54.649579
2      54.649579
3      53.882186
4      52.866866
         ...    
470     0.000000
342          NaN
344          NaN
361          NaN
396          NaN
Name: Distance, Length: 503, dtype: float64

In [61]:
vf_date_1165_43=vf[vf['Unit Time'].str.contains('2025/09/09 11:0')]
vf_date_1165_43.sort_values('Unit Time')

Row Id  Sequence #          Server Time            Unit Time  \
183  175743408994         NaN  2025/09/09 11:08:10  2025/09/09 11:00:19   
182  175743409018         NaN  2025/09/09 11:08:10  2025/09/09 11:02:19   
181  175743405125         NaN  2025/09/09 11:07:31  2025/09/09 11:03:52   
180  175743409046         NaN  2025/09/09 11:08:10  2025/09/09 11:04:19   
179  175743404797         NaN  2025/09/09 11:07:28  2025/09/09 11:07:06   
178  175743409073         NaN  2025/09/09 11:08:11  2025/09/09 11:07:10   
177  175743421209         NaN  2025/09/09 11:10:12  2025/09/09 11:08:39   
176  175743418806         NaN  2025/09/09 11:09:48  2025/09/09 11:09:10   

       Last Message Time  Latitude  Longitude  Driver Name         Event Name  \
183  2025/09/09 11:08:10  4.557641 -73.049324          NaN              Drive   
182  2025/09/09 11:08:10  4.566713 -73.032635          NaN              Drive   
181  2025/09/09 11:07:31  4.568895 -73.016848          NaN        No Seatbelt   
180  2025/09/09 11:08:10  4.569589 -73.012217          NaN              Drive   
179  2025/09/09 11:07:28  4.271055 -73.436302          NaN       AutoSnapshot   
178  2025/09/09 11:08:11  4.568523 -72.981169          NaN              Drive   
177  2025/09/09 11:10:12  4.567981 -72.967507          NaN  Collision Warning   
176  2025/09/09 11:09:48  4.568689 -72.966693          NaN              Drive   

                                 Raw Event  ... ECU Engine Hours + Off  \
183                         GpsData.Moving  ...                    NaN   
182                         GpsData.Moving  ...                    NaN   
181                CameraEvent.NO_SEATBELT  ...                    NaN   
180                         GpsData.Moving  ...                    NaN   
179                           AutoSnapshot  ...                    NaN   
178                         GpsData.Moving  ...                    NaN   
177  CameraEvent.FORWARD_COLLISION_WARNING  ...                    NaN   
176                         GpsData.Moving  ...                    NaN   

     Engine Odometer  Engine Odometer + Off  ECU Engine Odometer  \
183         21864.31                    NaN                  NaN   
182         21865.63                    NaN                  NaN   
181         21865.63                    NaN                  NaN   
180         21867.05                    NaN                  NaN   
179         21867.05                    NaN                  NaN   
178         21868.10                    NaN                  NaN   
177         22557.50                    NaN                  NaN   
176         21870.18                    NaN                  NaN   

     ECU Engine Odometer + Off HDOP VDOP RSSI  Battery Level  \
183                        NaN  NaN  NaN  -63            NaN   
182                        NaN  NaN  NaN  -63            NaN   
181                        NaN  0.8  NaN  -63            NaN   
180                        NaN  NaN  NaN  -63            NaN   
179                        NaN  0.6  NaN  -63            NaN   
178                        NaN  NaN  NaN  -63            NaN   
177                        NaN  0.7  NaN  -63            NaN   
176                        NaN  NaN  NaN  -63            NaN   

                                              Raw Data  
183  "{\"ingestVersion\":\"1.0.4\",\"linkedMode\":0...  
182  "{\"ingestVersion\":\"1.0.4\",\"linkedMode\":0...  
181  "{\"ingestVersion\":\"1.0.4\",\"linkedMode\":0...  
180  "{\"ingestVersion\":\"1.0.4\",\"linkedMode\":0...  
179  "{\"ingestVersion\":\"1.0.4\",\"linkedMode\":0...  
178  "{\"ingestVersion\":\"1.0.4\",\"linkedMode\":0...  
177  "{\"ingestVersion\":\"1.0.4\",\"linkedMode\":0...  
176  "{\"ingestVersion\":\"1.0.4\",\"linkedMode\":0...  

[8 rows x 50 columns]

21870.18=35196, este dato aparece el mileage report:

UPO081	9/09/2025 11:07	65, Barranca de Upía, Meta, 501007	3.35	347.82	35196.57
UPO081	9/09/2025 11:09	5127-553 65, Barranca, Meta, 501001	1165.43	1513.26	36362.01
UPO081	9/09/2025 11:12	65, Villanueva, Casanare, 855038	68.36	1581.62	36430.36
